lib/kafka_ex/abstract_socket.ex

```diff
@@ -0,0 +1,82 @@
+defmodule KafkaEx.AbstractSocket do
```

Comment:
Pretty much the only qualm I have here is the name. Sounds like Java ;-). `KafkaBrokerConnection` maybe? Something along the lines? Thanks for picking this up, by the way.